### DISCUSSION TITLE
Is it possible to run an onclick javascript function in a handlebars chart?

### DISCUSSION THREAD

**Discussion by asyncjason** [apache/superset#27431] [Q&A / Help]
Hi,

Is it possible to use an onclick javascript function in the handlebars chart?
I'd like my dashboard to be able to trigger an external REST API through a button click on my handlebars chart.

When I try this out I get react error:
```
react-dom.production.min.js:83 Uncaught Error: Minified React error #231; visit https://reactjs.org/docs/error-decoder.html?invariant=231&args[]=onClick&args[]=string for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
    at Rn (react-dom.production.min.js:83:506)
    at Hn (react-dom.production.min.js:84:47)
    at Yn (react-dom.production.min.js:84:373)
    at rt (react-dom.production.min.js:51:255)
    at Fn (react-dom.production.min.js:85:297)
    at Object.extractEvents (react-dom.production.min.js:117:45)
    at ft (react-dom.production.min.js:55:473)
    at H (react-dom.production.min.js:287:189)
    at F (react-dom.production.min.js:19:57)
    at Jt (react-dom.production.min.js:70:279)
```

https://react.dev/errors/231?invariant=231&args%5B%5D=onClick&args%5B%5D=string
Expected `onClick` listener to be a function, instead got a value of `string` type.

Cheers,

Jason

**Comment by mvsenin** [comment]
Hi,

Didn't you find any solution for that?

Had you look at the https://react.dev/reference/react-dom/components/common#mouseevent-handler - any success?

I'm stuck with that...

Thanx

**Comment by rusackas** [comment]
Currently, no, since that poses a variety of security risks.

**Reply by asyncjason** [reply]
It poses no risks in my use case though. Could it be added with a feature flag?

**Reply by rusackas** [reply]
Actually, I'm not 100% sure you can't... I might be wrong. You could try allowing script tags and all of that in your HTML_SANITIZATION_SCHEMA_EXTENSIONS, or disble HTML_SANITIZATION if you're really not worried about risks.

**Comment by gabgir** [comment]
Did you manage to make it work?
